src/PlatformEngines.jl

```diff
@@ -214,31 +214,46 @@ function probe_platform_engines!(;verbose::Bool = false)
     # package_opts_functor, list_opts_functor, parse_functor).  The probulator
     # will check each of them by attempting to run `$test_cmd`, and if that
     # works, will set the global compression functions appropriately.
-    gen_7z = (p) -> (unpack_7z(p), package_7z(p), list_7z(p), parse_7z_list)
+    gen_7z = (p) -> (unpack_7z(p), package_7z(p), list_7z(p), parse_7z_list, r"Path = ([^\r\n]+)\r?\n(?:[^\r\n]+\r?\n)+Symbolic Link = ([^\r\n]+)"s)
     compression_engines = Tuple[]
 
+    write("demoTar.txt","Demo file for tar listing")
     for tar_cmd in [`tar`, `busybox tar`]
+        # try to determine the tar list format
+        local symlink_parser
+        try
+            tarListing = read(pipeline(`$tar_cmd -c demoTar.txt`,`$tar_cmd -tv`), String)
+            m = match(r"((?:\S+\s+)+?)demoTar\.txt", tarListing)[1]
+            nargs = length(split(m, " "; keepempty = false))
+            symlink_parser = Regex("^l(?:\\S+\\s+){$nargs}(.+?)(?: -> (.+?))?\\r?\$", "m")
+        catch
+            # generic expression for symlink parsing
+            # this will fail, if the symlink contains space characters (which is highly improbable, though)
+            symlink_parser = r"^l.+? (\S+?)(?: -> (.+?))?\r?$"m
+        end
```

Comment:
It would be very helpful to bake into this code, with comments, the different kinds of text the regexes are supposed to be matching.  E.g. capture the output across the different versions of `tar` that we are matching against, so we can see what the Regexes are doing.

Reply:
I see what you added, and that's great, but I literally meant examples of the text you're matching.  Something like:

```
# this regex matches the line "l. foo -> foo"
...
# this regex matches the line "l  foo -> foo"
```

Just so that we can see examples of what the lines look like.